docs/dev/javascript-bundling.md

```diff
@@ -0,0 +1,41 @@
+# Javascript bundling
+
+We use esbuild to bundle our custom Javascript into single files for each kind
+of page. For example, a point location page would be served `point.js` which
+has been bundled to include all of the code necessary to display that page
+propertly. This allows us to minimize the number of network calls required to
+load the page as well as giving us control over minification.
+
+We use ESM modules for our custom code. This helps create clean scope boundaries
+so we don't have to worry about polluting the global scope. It also helps us
+manage import order, since esbuild can just figure it out for us.
+
+## Drupal libraries
+
+We define what Javascript to deploy in the Drupal theme's libraries file at
+`web/themes/new_weather_theme/new_weather_theme.libraries.yml`. For custom code,
+there should be one entry per page type. The same code may be bundled in multiple
+page types, but we only want to deliver a single bundled file per page type.
```

Comment:
Can we add something about the fact that we need the yml key to end in `-page` to actually bundle?

Reply:
Yes, thank you! I was struggling to remember what needed to be documented. Slow start with the Monday brain and a gazillion meetings. 😬